### PR TITLE
Martyr Cyborg Module, borg law 2 explode

### PR DESCRIPTION
**Pull request by Mangohydra**
## About the PR
Syndicate agents can now purchase a cyborg module that gives emagged borgs a suicide bombing button only for 4 tc (emag not included)

## Why
this was entirely for shits and giggles, and later people actually liked it so i turned it into an actual PR
The module includes a single explosive timed for 4.5 seconds, it will gib (you and) explosive resistant targets granted you are right next to them, its only one use so use it wisely.

## Media
uplink image
borgbomb

the very cool module sprite
image

module in use
https://github.com/space-wizards/space-station-14/assets/156087924/a8cebd3e-143f-44c3-88d9-db38aee36ef0
(blame the 10mb upload limit for the 480p video)

**Changelog**

🆑

  -  add: Added a new syndicate cyborg module that contains a very dangerous explosive as a 4tc purchase.

### CODE REVIEW

**Comment by TheShuEd** [comment]
Add audio licenses please

**Comment by Dutch-VanDerLinde** [comment]
changelog borken.. you need 🆑 and the syntax is incorrect

**Comment by Mangohydra** [comment]
don't worry i got this trust me that failure will be gone in no time

**Comment by Dutch-VanDerLinde** [comment]
changelog emoji good but syntax still bad, this is what it should look like:

:cl:
- add: a martyr cyborg module and the fun explosive included with it, 8tc purchase
- add: more paranoia (that borg over there might be a bomb you never know)

**Comment by Mangohydra** [comment]
> changelog emoji good but syntax still bad, this is what it should look like:
> 
> 🆑
> 
>     * add: a martyr cyborg module and the fun explosive included with it, 8tc purchase
> 
>     * add: more paranoia (that borg over there might be a bomb you never know)

thank you i'll be adding that now (i forgor)

**Comment by DangerRevolution** [comment]
emag to emag Borg = 8 TC
suicide module = 8 TC

Unless the Borg (another player you have to rely on) is a aim-God and they gib your target, you're potentially spending 80% of your TC to crit your target and alert the whole station.

TLDR: Is this too expensive?

**Comment by Fildrance** [comment]
> emag to emag Borg = 8 TC suicide module = 8 TC
> 
> Unless the Borg (another player you have to rely on) is a aim-God and they gib your target, you're potentially spending 80% of your TC to crit your target and alert the whole station.
> 
> TLDR: Is this too expensive?

It can be even underpriced! explosion power is great and borg have most of the time more access then everyone and good speed, no slowdown. It can be OP, but totally not too expensive - its almost 100% kill it feels so far - we need more game testing before re-evaluating price.

**Comment by Simyon264** [comment]
This definitely is too expensive considering that the borg can rat you out afterwards.

>get emaged
>blow up target
>laws are gone
>rat out owner to not get wiped as posi/brain removed as mmi

There are better options for gibbing targets that cost way less.

**Comment by deltanedas** [comment]
baseball bat 0 tc
2 fish 4 tc
minibomb uh 6 tc i think
emag recycler 8 tc and it cant have skill issue

**Comment by Simyon264** [comment]
C4 - 2tc
Remote signaller - 0tc
Minibomb - 4tc
Survival box - 0tc

Those combined are a fun way to remotly gib your target and it only costs 6 TC!

**Comment by Fildrance** [comment]
Borg have special access and can follow target. Thrown object can meet obstructions or you can just miss... and i think minibomb area is much smaller. I have to also remind that there won't be any steel proof that you are part of explosion business. If there is A LOT of cases when borg brain will rat you out - then we will fix it by assploding brain with borg. Zero evidence, you are clean to go home... you can orchestrate explosion on other side of station while sitting and drinking on the beach.

**Comment by Simyon264** [comment]
> Borg have special access and can follow target. Thrown object can meet obstructions or you can just miss... and i think minibomb area is much smaller. I have to also remind that there won't be any steel proof that you are part of explosion business. If there is A LOT of cases when borg brain will rat you out - then we will fix it by assploding brain with borg. Zero evidence, you are clean to go home... you can orchestrate explosion on other side of station while sitting and drinking on the beach.

Exploding the brain of the borg is bad. Forced round removal where the only way to not get round removed is to break your laws is a bad game mechanic.

**Comment by Mangohydra** [comment]
going to tweak some shit and lower the price, its gonna be 5 tc just like the weapons module, working on the changes now

**Comment by vorkathbruh** [comment]
make it 6 tc so you buy 2 of these and an emag :trollface:

File: Resources/Prototypes/Entities/Objects/Specific/Robotics/borg_modules.yml [hunk 1]
```diff
@@ -554,3 +554,16 @@
       items:
         - WeaponLightMachineGunL6C
         - PinpointerSyndicateNuclear
+
+- type: entity
+  id: BorgModuleMartyr
+  parent: [ BaseBorgModule, BaseProviderBorgModule ]
+  name: Martyr Cyborg Module
```
Comment (TheShuEd) [inline Resources/Prototypes/Entities/Objects/Specific/Robotics/borg_modules.yml:561]: i think u need add description

File: Resources/Prototypes/Entities/Objects/Weapons/Throwable/grenades.yml [hunk 2]
```diff
@@ -120,6 +120,29 @@
       params:
         volume: 12
 
+- type: entity
+  name: Self Destruct
```
Comment (TheShuEd) [inline Resources/Prototypes/Entities/Objects/Weapons/Throwable/grenades.yml:124]: add noSpawn: true

File: Resources/Prototypes/Entities/Objects/Specific/Robotics/borg_modules.yml (hunk 1 above)
Comment (Mangohydra) [reply Resources/Prototypes/Entities/Objects/Specific/Robotics/borg_modules.yml:561]: finished
image

File: Resources/Prototypes/Entities/Objects/Weapons/Throwable/grenades.yml (hunk 2 above)
Comment (Mangohydra) [reply Resources/Prototypes/Entities/Objects/Weapons/Throwable/grenades.yml:124]: got this one too
image

**Comment by SoulFN** [comment]
I think this should be an action, like minibomb implant

File: Resources/Prototypes/Entities/Objects/Weapons/Throwable/grenades.yml [hunk 3]
```diff
@@ -120,6 +120,30 @@
...
+  description: "Go out on your own terms, hell, even take them with you! kill them all, you know you want to! do it! (disclaimer: does not arm without power)"
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Throwable/grenades.yml:125]:
```suggestion
  description: "Go out on your own terms!
```

Way too long and fluffy, also capitalisation.

File: Resources/Prototypes/Entities/Objects/Weapons/Throwable/grenades.yml [hunk 4]
```diff
@@ -120,6 +120,30 @@
...
+  parent: GrenadeBase
+  id: SelfDestructSeq
+  noSpawn: true
+  components:
+  - type: ExplodeOnTrigger
+  - type: Explosive
+    explosionType: Minibomb
+    totalIntensity: 800
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Throwable/grenades.yml:133]:
```suggestion
    totalIntensity: 400
```

Instagibbing is a little ridiculous

File: Resources/Audio/Effects/Grenades/SelfDestruct/attributions.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+- files:
+    - SDS_Charge2.ogg
+  license: CC-BY-SA-3.0
+  source: https://freesound.org/people/Teh_Bucket/sounds/518739/
+  # couldn't figure out how to source multiple (the right way) without shit breaking so heres the rest: https://pixabay.com/sound-effects/switchbigpowerwav-14710/ https://pixabay.com/sound-effects/shield-recharging-107016/
```
Comment (metalgearsloth) [inline Resources/Audio/Effects/Grenades/SelfDestruct/attributions.yml:5]:
```suggestion
  license: Custom
  source: https://freesound.org/people/Teh_Bucket/sounds/518739/
  # couldn't figure out how to source multiple (the right way) without shit breaking so heres the rest: https://pixabay.com/sound-effects/switchbigpowerwav-14710/ https://pixabay.com/sound-effects/shield-recharging-107016/
```

CC 3.0 is incorrect as pixabay is not licensed under that.

File: Resources/Audio/Effects/Grenades/SelfDestruct/attributions.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+- files:
+    - SDS_Charge2.ogg
```
Comment (mirrorcult) [inline Resources/Audio/Effects/Grenades/SelfDestruct/attributions.yml:2]:
```suggestion
    - SDS_Charge.ogg
    - SDS_Charge2.ogg
```

File: Resources/Prototypes/Entities/Objects/Specific/Robotics/borg_modules.yml [hunk 7]
```diff
@@ -554,3 +554,17 @@
...
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Specific/Robotics/borg_modules.yml:561]:
```suggestion
  name: martyr cyborg module
```

File: Resources/Prototypes/Entities/Objects/Weapons/Throwable/grenades.yml [hunk 8]
```diff
@@ -120,6 +120,30 @@
       params:
         volume: 12
 
+- type: entity
+  name: Self Destruct
+  description: "Go out on your own terms!
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Weapons/Throwable/grenades.yml:125]:
```suggestion
  description: Go out on your own terms!
```

File: Resources/Prototypes/Entities/Objects/Specific/Robotics/borg_modules.yml [hunk 9]
```diff
@@ -554,3 +554,17 @@
...
+  description: "A module that comes with an explosive you probably don't want you to handle."
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Specific/Robotics/borg_modules.yml:562]:
```suggestion
  description: A module that comes with an explosive you probably don't want to handle yourself.
```

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Review by mirrorcult — APPROVED** [review]
